src/lib/helpers/youtubeTranscriptsHandling.ts

```diff
@@ -26,18 +27,18 @@ const ESCAPE_SUBSTITUTIONS = {
 export async function handleTranscripts(
     innertubeClient: Innertube,
     videoId: string,
-    // @ts-ignore to be fixed
-    selectedCaption,
+    selectedCaption: CaptionTrackData,
 ) {
     const lines: string[] = ["WEBVTT"];
 
     const info = await innertubeClient.getInfo(videoId);
     const transcriptInfo = await (await info.getTranscript()).selectLanguage(
-        selectedCaption.name.simpleText,
+        selectedCaption.name.text || "",
     );
-    const rawTranscriptLines =
-        // @ts-ignore to be fixed
-        transcriptInfo.transcript.content.body.initial_segments;
+    const rawTranscriptLines = transcriptInfo.transcript.content?.body
+        ?.initial_segments;
+
+    if (rawTranscriptLines == undefined) throw new HTTPException(404);
```

Comment:
Maybe we could send a specific error to the user? Like:
```suggestion
    if (rawTranscriptLines == undefined) {
        throw new HTTPException(404, {
           res: new Response("No captions for this video ID."),
        });
    }
```